File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int ans = 0;
        int zeroes = 0;
        int ones = (int) s.chars().filter(ch->ch == '1').count();

        for (int i = 0; i < s.length()-1; i++) {
            zeroes += s.charAt(i) == '0' ? 1 : 0;
            ones += s.charAt(i) == '1' ? -1 : 0;
            ans = Math.max(ans, zeroes+ones);
        }

        return ans;
        
    }
}

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        ans = 0
        ones = s.count("1")
        zeros = 0
        for i in range(len(s)-1):
            char = s[i]
            if char == "0":
                zeros += 1
            else:
                ones -= 1
            ans = max(ans, ones+zeros)
        return ans
                 

            
        